### ISSUE TITLE
2.1.0 regression when using `all=True` on project.mergerequests.list()

### ISSUE THREAD

**Issue by dmerejkowsky** [python-gitlab/python-gitlab#1044]
## Description of the problem, including code/CLI snippet

```python
import os
import dotenv
from gitlab import Gitlab


dotenv.load_dotenv()
gitlab_url = os.environ["GITLAB_URL"]
gitlab_token = os.environ["GITLAB_TOKEN"]
project_name = os.environ["PROJECT_NAME"]

gitlab = Gitlab(gitlab_url, private_token=gitlab_token)
project = gitlab.projects.get(project_name)
all_mrs = project.mergerequests.list(state="opened", all=True")
print(all_mrs)
```

## Expected Behavior

The list of all opened merge requests is returned

## Actual Behavior

Code crashes with:

```
Traceback (most recent call last):
  File "/home/dmerej/3rdpart/python-gitlab/gitlab/exceptions.py", line 271, in wrapped_f
    return f(*args, **kwargs)
  File "/home/dmerej/3rdpart/python-gitlab/gitlab/mixins.py", line 141, in list
    obj = self.gitlab.http_list(path, **data)
  File "/home/dmerej/3rdpart/python-gitlab/gitlab/__init__.py", line 651, in http_list
    return list(GitlabList(self, url, query_data, **kwargs))
  File "/home/dmerej/3rdpart/python-gitlab/gitlab/__init__.py", line 778, in __init__
    self._query(url, query_data, **kwargs)
  File "/home/dmerej/3rdpart/python-gitlab/gitlab/__init__.py", line 783, in _query
    result = self._gl.http_request("get", url, query_data=query_data, **kwargs)
  File "/home/dmerej/3rdpart/python-gitlab/gitlab/__init__.py", line 573, in http_request
    response_body=result.content,
gitlab.exceptions.GitlabHttpError: 400: order_by does not have a valid value

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "bug.py", line 13, in <module>
    all_mrs = project.mergerequests.list(state="opened", all=True)
  File "/home/dmerej/3rdpart/python-gitlab/gitlab/exceptions.py", line 273, in wrapped_f
    raise error(e.error_message, e.response_code, e.response_body)
gitlab.exceptions.GitlabListError: 400: order_by does not have a valid value
```

## Specifications

  - python-gitlab version: 6c5458a3bfc3208ad2d7cc40e1747f7715abe449
  - API version you are using (v3/v4): v4
  - Gitlab server version (or gitlab.com):  12.8.2-ee 

## Notes:

* The code used to work with 2.0.0 release

**Comment by dmerejkowsky** [comment]
_Update_ :

* reverting 99b4484da924f9378518a1a1194e1a3e75b48073 fixes the issue
* Specifying `order_by` also works: `all_mrs = project.mergerequests.list(state="opened", all=True, order_by="created_at")`
* Some "list" GitLab API allow sorting by IDs, and other by date, which is probably why the bug was not caught
* Maybe it's enough to update the changelog?

**Comment by max-wittig** [comment]
Sorry for the breakage. I guess some endpoints don't allow sorting by id  ☹️ 

I would say we revert it for now.

**Comment by max-wittig** [comment]
I've published 2.1.2 to revert this. Sorry again!

**Comment by dmerejkowsky** [comment]
Wow that was fast :) Thanks!
